src/rpc/rpcHandler.ts

```diff
@@ -141,195 +130,79 @@ export class RpcHandler {
-                entryPoint
-            })
-
-        if (
-            userOperationNonceValue >
-            currentNonceValue + BigInt(queuedUserOperations.length)
-        ) {
-            this.mempool.add(userOperation, entryPoint)
-            this.eventManager.emitQueued(opHash)
-            return "queued"
-        }
-
-        if (this.config.dangerousSkipUserOperationValidation) {
-            const [success, errorReason] = await this.mempool.add(
-                userOperation,
-                entryPoint
-            )
-            if (!success) {
-                this.eventManager.emitFailedValidation(
-                    opHash,
-                    errorReason,
-                    getAAError(errorReason)
-                )
-                throw new RpcError(errorReason, ValidationErrors.InvalidFields)
-            }
-            return "added"
-        }
-
-        if (apiVersion !== "v1") {
-            await this.validator.validatePreVerificationGas({
-                userOperation,
-                entryPoint
-            })
+            ]
         }
-        await this.mempool.checkEntityMultipleRoleViolation(
-            entryPoint,
-            userOperation
-        )
-
-        const validationResult = await this.validator.validateUserOperation({
-            userOperation,
-            queuedUserOperations,
-            entryPoint
-        })
-
-        await this.reputationManager.checkReputation(
-            userOperation,
-            entryPoint,
-            validationResult
-        )
-
-        await this.mempool.checkEntityMultipleRoleViolation(
-            entryPoint,
-            userOperation
-        )
 
-        const [success, errorReason] = await this.mempool.add(
-            userOperation,
-            entryPoint,
-            validationResult.referencedContracts
-        )
-
-        if (!success) {
-            this.eventManager.emitFailedValidation(
-                opHash,
-                errorReason,
-                getAAError(errorReason)
-            )
-            throw new RpcError(errorReason, ValidationErrors.InvalidFields)
-        }
-        return "added"
+        return [true, ""]
     }
 
     async validateEip7702Auth({
         userOperation,
         validateSender = false
-    }: { userOperation: UserOperation; validateSender?: boolean }) {
+    }: { userOperation: UserOperation; validateSender?: boolean }): Promise<
+        [boolean, string]
+    > {
         if (!userOperation.eip7702Auth) {
-            throw new RpcError(
-                "UserOperation is missing eip7702Auth",
-                ValidationErrors.InvalidFields
-            )
+            return [true, ""]
         }
 
         if (!this.config.codeOverrideSupport) {
-            throw new RpcError(
-                "eip7702Auth is not supported on this chain",
-                ValidationErrors.InvalidFields
-            )
+            return [false, "eip7702Auth is not supported on this chain"]
```

Comment:
we can also parallelise `recoverAuthorizationAddress`, `nonceOnChain` and `this.config.publicClient.getCode`